tensorflow/compiler/xla/service/gpu/gpu_executable.cc

```diff
@@ -72,6 +75,42 @@ GpuExecutable::~GpuExecutable() {
                                                assignment_);
 }
 
+Status GpuExecutable::CheckCompatibilityWithServiceExecutableRunOptions(
+    const ServiceExecutableRunOptions* run_options) {
+  se::Stream* main_stream = run_options->stream();
+
+  string vendor = main_stream->parent()->GetDeviceDescription().device_vendor();
+
+  if (vendor == "Advanced Micro Devices, Inc") {
```

Comment:
would it be a good idea to extract these strings from `cuda_gpu_executor.cc` and `rocm_gpu_executor.cc` into a header file?

Reply:
Agreed. This thought passed my mind as well. The file being touched then will be on `stream_executor` side. I can file another PR just for the naming concerns and take the opportunity to see if this raw string is used in other places as well.

Reply:
actually use a switch case on `main_stream->parent()->platform_kind()` is even better.